src/zb_links/api/link/helpers/link_helpers.py

```diff
@@ -4,13 +4,58 @@
 import re
 
 
-def update_set_by_intersect(old_set, new_set):
-    if not old_set:
-        return new_set
-    return set.intersection(old_set, new_set)
+def get_new_id():
+    """
+
+    Returns
+    -------
+    new_link_id: int
+        an id number not taken up by another link in
+        the document_external_ids table.
+
+    """
+    connection = db.engine.connect()
+    max_request = "SELECT MAX(id) FROM document_external_ids;"
+    max_id = connection.execute(max_request).fetchone()[0]
+    new_link_id = max_id + 1
```

Comment:
remove the variable new_link_id

Reply:
To be more precise: I would merge the statement in line 20 and 21 and just write. https://techbeacon.com/app-dev-testing/why-unnecessary-variables-are-bad-your-code I would see this a "unnecessary variable before return"  problem not less and not more.

Reply:
we will look into how the id's are generated

Reply:
this is an orthogonal problem. See my comment below.